### PR TITLE
16078 make GraphQL NumberFilter optional

### LINKED ISSUE #16078: GraphQL: Required fields that probably shouldn't be
### Deployment Type

Self-hosted

### NetBox Version

v4.0.1

### Python Version

3.11

### Steps to Reproduce

1. Go to GraphQL console
2. Enter this query
```graphql
query {
  ip_address_list(filters: {tenant: "my-tenant"}){
    address
  }
}
```
3. You'll have an error that some required fields are missing

### Expected Behavior

The `family`, `mask_length__gte` and `mask_length__lte` filters shouldn't be mandatory.

Probably the same for other filters.
Also broken on https://demo.netbox.dev/

### Observed Behavior

```json
{
  "data": null,
  "errors": [
    {
      "message": "Field 'IPAddressFilter.family' of required type 'Int!' was not provided.",
      "locations": [
        {
          "line": 2,
          "column": 28
        }
      ]
    },
    {
      "message": "Field 'IPAddressFilter.mask_length__gte' of required type 'Int!' was not provided.",
      "locations": [
        {
          "line": 2,
          "column": 28
        }
      ]
    },
    {
      "message": "Field 'IPAddressFilter.mask_length__lte' of required type 'Int!' was not provided.",
      "locations": [
        {
          "line": 2,
          "column": 28
        }
      ]
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by arthanson**
### Fixes: #16078 

Make NumberFilter not required in GraphQL.

### CODE REVIEW

**Review by jeremystretch — CHANGES_REQUESTED** [review]
@arthanson could you add a few basic GrapQL filter tests to catch this sort of error?

**Comment by arthanson** [comment]
@jeremystretch added test used graphql_filter in test-case so it can be extended.  Kept this current test narrow in scope as there is #15569 that can expand on this in the future - will also want to look at adding lookups to this in the future.

File: netbox/utilities/testing/api.py [hunk 1]
```diff
@@ -440,13 +440,8 @@ def _get_graphql_base_name(self):
             base_name = self.model._meta.verbose_name.lower().replace(' ', '_')
             return getattr(self, 'graphql_base_name', base_name)
 
-        def _build_query(self, name, **filters):
+        def _build_query_with_filter(self, name, filter_string):
```
Comment (jeremystretch) [inline netbox/utilities/testing/api.py:443]: Could you add docstrings for these three methods please? They all look very similar and it's not immediately clear how they differ.

- `_build_query_with_filter()`
- `_build_filtered_query()`
- `_build_query()`

**Review by jeremystretch — CHANGES_REQUESTED** [review]

File: netbox/utilities/testing/api.py (hunk 1 above)
Comment (arthanson) [reply netbox/utilities/testing/api.py:443]: done

**Review by jeremystretch — APPROVED** [review]
Thanks @arthanson!

**Pull request merged**